plugin/etcd/plugin.go

```diff
@@ -0,0 +1,451 @@
+		Author:  "Jason Zhou, Pururva Lakkad, Naveed Ahmad, Sriniketh Varma Dasarraju",
+		Version: "0.0.1",
+		Features: plugin.PluginFeatures{
+			Target: "yes",
+			Store:  "no",
+		},
+		Example: `
+			{
+			"endpoints" : "https://192.168.42.45:2379"   # REQUIRED
+
+			"auth"  : false												# is role based or cert based auth enabled on the etcd cluster
+			"user"  : "admin",            								# username for role based authentication
+			"pass"  : "p@ssw0rd"          								# password for role based authentication
+			"clientCertPath" : "/tmp/test-certs/test-client.crt" 		# path to client certificate
+			"clientKeyPath"  : "/tmp/test-certs/test-client-key.key" 	# path to client key
+			"caCertPath"     : "/tmp/test-certs/test-ca.crt" 			# path to CA certificate
+			"fullOverwrite"  : "false"								    # enable or disable full overwrite of the cluster
+			"prefix"		 : "starkandwayne/"							# backup specific keys
+			}
+			`,
+		Defaults: `
+			{
+			}
+			`,
+		Fields: []plugin.Field{
+			plugin.Field{
+				Mode:     "target",
+				Name:     "endpoints",
+				Type:     "string",
+				Title:    "Endpoint",
+				Help:     "IP addresses of the etcd nodes in the cluster",
+				Example:  "https://192.168.42.45:2379",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "auth",
+				Type:    "bool",
+				Title:   "Authentication",
+				Help:    "Is role based or cert based authentication enabled on the etcd cluster",
+				Example: "false",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "user",
+				Type:    "string",
+				Help:    "Username for role based authentication",
+				Title:   "Username",
+				Example: "admin",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "pass",
+				Type:    "password",
+				Help:    "Password for role based authentication",
+				Title:   "Password",
+				Example: "p@ssw0rd",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "clientCertPath",
+				Type:    "string",
+				Help:    "Path to the certificate issued by the CA for the client connecting to the ETCD cluster",
+				Title:   "Client Certificate File Path",
+				Example: "/tmp/test-certs/test-client.crt",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "clientKeyPath",
+				Type:    "string",
+				Help:    "Path to the key issued by the CA for the client connecting to the ETCD cluster",
+				Title:   "Client Key File Path",
+				Example: "/tmp/test-certs/test-client-key.key",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "caCertPath",
+				Type:    "string",
+				Help:    "Path to the CA certificate that issued the client cert and key",
+				Title:   "Trusted CA File Path",
+				Example: "/tmp/test-certs/test-ca.crt",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "fullOverwrite",
+				Type:    "bool",
+				Help:    "If this is enabled, the existing key/value pairs will be deleted. The values will be restored using the backup archive.",
```

Comment:
It's worth pointing out here that only keys underneath `prefix` will be deleted.